google/resource_manager_operation.go

```diff
@@ -0,0 +1,46 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+package google
+
+import (
+	"fmt"
+)
+
+type ResourceManagerOperationWaiter struct {
+	Config *Config
+	CommonOperationWaiter
+}
+
+func (w *ResourceManagerOperationWaiter) QueryOp() (interface{}, error) {
+	if w == nil {
+		return nil, fmt.Errorf("Cannot query operation, it's unset or nil.")
+	}
+	// Returns the proper get.
+	url := fmt.Sprintf("https://cloudresourcemanager.googleapis.com/v1/%s", w.CommonOperationWaiter.Op.Name)
+	return sendRequest(w.Config, "GET", url, nil)
+}
+
+func resourceManagerOperationWaitTime(config *Config, op map[string]interface{}, project, activity string, timeoutMinutes int) error {
```

Comment:
`op` here should probably be an interface or a struct. Maps that have 'special' keys are dangerous.

Reply:
This is an MM-generated operation handler, so this is expected.

Reply:
I ah see why we can't use a ResourceManagerOp, but we could still use a more generic Operation interface. It is strange to take a struct and 'dynamically' cast it to json, then to a map, then back into a struct with the same structure.

Reply:
Talked to Chris offline- this looks especially silly because we're in a transitional phase, but after a few PRs land it should make more sense. Once we're closer to a full MM implementation we'll reevaluate his suggestion.